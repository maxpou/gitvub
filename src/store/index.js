import Vue from 'vue'
import Vuex from 'vuex'
import repositories from './modules/repositories'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

const store = new Vuex.Store({
  modules: {
    repositories
  },
  strict: debug
})

/* istanbul ignore if */
if (module.hot) {
  module.hot.accept([
    './modules/repositories'
  ], () => {
    const repositories = require('./modules/repositories').default
    store.hotUpdate({
      modules: {
        repositories
      }
    })
  })
}

export default store
