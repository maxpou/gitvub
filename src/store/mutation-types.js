export const RECEIVE_REPOSITORIES = 'RECEIVE_REPOSITORIES'
export const RECEIVE_LANGUAGES = 'RECEIVE_LANGUAGES'
export const FETCH_USER_REPOS = 'FETCH_USER_REPOS'
export const FETCH_REPO_DETAIL = 'FETCH_REPO_DETAIL'
export const RECEIVE_REPOSITORY_LANGUAGE = 'RECEIVE_REPOSITORY_LANGUAGE'
export const RECEIVE_REPOSITORY_README = 'RECEIVE_REPOSITORY_README'
