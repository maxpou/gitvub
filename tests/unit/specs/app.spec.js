import { mount } from 'vue-test-utils'
import MainComponent from '@/App.vue'
import store from '@/store'

describe('App.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(MainComponent, { store })
    const appTemplate = wrapper.html()

    expect(appTemplate).toMatchSnapshot()
  })
})
