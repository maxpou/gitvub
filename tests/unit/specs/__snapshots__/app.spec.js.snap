// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue test initial rendering 1`] = `"<div id=\\"app\\" class=\\"container-fluid\\"><router-view></router-view></div>"`;

exports[`App.vue test rendering with datas 1`] = `"<div id=\\"app\\" class=\\"container-fluid\\"><router-view></router-view></div>"`;
